crates/flox-rust-sdk/src/models/environment/managed_environment.rs

```diff
@@ -414,6 +412,33 @@ impl ManagedEnvironment {
     }
 }
 
+fn lock_env(
```

Comment:
nit nonblocking: this function name isn't super well differentiated from `lock_manifest`. Idk if there's something better, maybe `lock_generation`

Reply:
`lock_env_pointer`?
`lock_env_descriptor`?

proposal to change `pointer` to `descriptor` :D